### PR TITLE
Add docs for robot accounts

### PR DESCRIPTION
**Pull request by a-mccarthy**
Signed-off-by: Abigail McCarthy <mabigail@vmware.com>

### CODE REVIEW

**Comment by a-mccarthy** [comment]
Open questions:
* Are the following still true? They are writing in the current versions of the docs, but i'm not sure if they still apply to the new robot accounts
  * Harbor does not store robot account secrets, so you must either download the secret or copy and paste its contents into a text file. There is no way to get the secret from Harbor after you have created the robot account.
  * Robot Accounts cannot log in to the Harbor interface.
  * Robot Accounts can only perform operations by using the Docker and Helm CLIs.
* What happens when you enable/disable accounts, anything someone should know about in advance?
* Do we want to document any examples of using a robot account to perform an action via the API?

File: docs/administration/robot-accounts/_index.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+1. Go to **Configuration** and select **System Settings**.
+1. In the **Robot Token Expiration (Days)** row, modify the number of days after which robot account tokens expire.
+
+    ![Set robot account token expiry](../../img/set-robot-account-token-duration.png)
+
+## Configure Robot Account Prefix
+
+By default, robot account names use a prefix of `robot$`. Harbor uses this prefix to distinguish a robot account from a user account. The full name of a robot account is the prefix and the name you provided when creating the robot account. For example if you create a new robot account with the name `test`, the full name is  `robot$test`.
+
+The same prefix is used for all robot accounts, including both system and project robot accounts. When you update this value, it will apply to all existing and future system and project robot accounts, including robot accounts created in Harbor v2.1 and earlier.
```
Comment (wy65701436) [inline docs/administration/robot-accounts/_index.md:125]: The prefix will not be updated for the robots created at v2.1 or before. The prefix editing only works for v2.2+ robots.

**Comment by wy65701436** [comment]
> Open questions:
> 
>     * Are the following still true? They are writing in the current versions of the docs, but i'm not sure if they still apply to the new robot accounts
>       
>       * Harbor does not store robot account secrets, so you must either download the secret or copy and paste its contents into a text file. There is no way to get the secret from Harbor after you have created the robot account.
>       * Robot Accounts cannot log in to the Harbor interface.
>       * Robot Accounts can only perform operations by using the Docker and Helm CLIs.
> 
>     * What happens when you enable/disable accounts, anything someone should know about in advance?
> 
>     * Do we want to document any examples of using a robot account to perform an action via the API?

1, yes, Harbor doesn't store the secret of robots, so user has to copy/paste it after it created into their env.
2, Robot is designed to access Harbor with oci client or API only, cannot login to UI.
3, Robot can perform operations by oci client or RSAT API.
4, Once it disabled, when oci client uses the robot to login, it will get a 401. User can get an error in the core.log, like `failed to authenticate disabled robot account: robot$demo`
5, The API access with a robot has no different from a user, the swagger doc is enough I believe.

**Review by wy65701436 — APPROVED** [review]
lgtm

File: docs/administration/robot-accounts/_index.md (hunk 1 above)
Comment (a-mccarthy) [reply docs/administration/robot-accounts/_index.md]: Thank you, i updated this information.

File: docs/administration/robot-accounts/_index.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+## View System Robot Accounts
+
+1. Log into your harbor instance as an administrator.
+1. Go to **Robot Accounts** item under **Administration**.
+
+![System robot account page](../../img/system-robot-account-page.png)
+
+This page lists all available system robot accounts for your Harbor instance. The table lists the following information for each system robot account,
+
+* The name of a system account. This is derived from robot account prefix configured for your Harbor instance and the name assigned to the account when it was created. A robot account name follows thee format `<prefix><account_name>`. If you use the search function on this page, you only need to search for the account name without the prefix.
```
Comment (AllForNothing) [inline docs/administration/robot-accounts/_index.md:34]: typo :  "thee"

File: docs/administration/robot-accounts/_index.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    ![View list of all projects associated with a system robot account](../../img/list-robot-account-projects.png)
+
+* The created time shows when the robot account was created.
+* The expires at time of the account which is calculated based on the created time and the expiration time set when creating the project robot account.
+* The description of the system robot account.
+
+
+## Add a System Robot Account
+
+1. Log in to the Harbor interface with an account that has at least project administrator privileges.
```
Comment (AllForNothing) [inline docs/administration/robot-accounts/_index.md:47]: Only system admin can add a  system robot account

Comment (AllForNothing) [inline docs/administration/robot-accounts/_index.md:47]: At least system admin?

File: docs/administration/robot-accounts/_index.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    By default the table shows all projects in your Harbor instance. You are able to filter for projects using the **filter icon** to the right of Project Name header. Note that the project table may be broken into pages and only display a subset of projects at one time depending on how many project you have in your Harbor instance and how many project match your filter criteria.
+
+      ![Filter project names](../../img/robot-account-filter-project-name.png)
+
+    Use the **Permissions** dropdown to choose which permissions to assign to a particular project. You are able to control which permissions to assign to an individual robot account by project, allowing you fine grained control over each robot account.
+
+    ![](../../img/set-robot-account-permissions.png)
+
+    Click the **Rest Permissions** dropdown to control which permissions are available for each project. Selecting or unselecting a permission will add or remove the permission for every project. Using this option will adjust permissions for all projects, not just the projects shown if you have filtered for a specific project name.
```
Comment (AllForNothing) [inline docs/administration/robot-accounts/_index.md:71]: Typo: reset

File: docs/administration/robot-accounts/_index.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+    ![Reset robot account permissions](../../img/reset-robot-account-permissions.png)
+
+    Click **Select All** to associate thee system robot account with all of the projects shown in the table. If you are filtering by project name, this option will only select the filtered projects.
```
Comment (AllForNothing) [inline docs/administration/robot-accounts/_index.md:75]: Typo: thee system robot

File: docs/working-with-projects/project-configuration/create-robot-accounts.md [hunk 6]
```diff
@@ -1,62 +1,115 @@
...
+
+## View Project Robot Accounts
 
 1. Log in to the Harbor interface with an account that has at least project administrator privileges.
 1. Go to **Projects**, select a project, and select **Robot Accounts**.
 
-    ![Robot accounts](../../../img/add-robot-account.png)
+![Project robot accounts page](../../../img/project-robot-account.png)
+
+This page lists all available system robot accounts for a project. The table lists the following information for each robot account,
```
Comment (AllForNothing) [inline docs/working-with-projects/project-configuration/create-robot-accounts.md:34]: list  all project robot accounts

Comment (a-mccarthy) [reply docs/working-with-projects/project-configuration/create-robot-accounts.md]: updated

**Comment by a-mccarthy** [comment]
Thanks for reviewing @AllForNothing! I made the updates from your comments

**Review by AllForNothing — APPROVED** [review]
LGTM

**Review by xaleeks — APPROVED** [review]

**Comment by xaleeks** [comment]
I'm merging.  We can always update later.  Thanks Abbie

**Pull request merged**